cpp/src/copying/gather.cu

```diff
@@ -43,50 +42,39 @@ struct dispatch_map_type {
                    "Index out of bounds.");
     }
 
-    if (allow_negative_indices) {
-      destination_table =
-        gather(source_table,
-               thrust::make_transform_iterator(gather_map.begin<map_type>(),
-                                               index_converter<map_type>{source_table.num_rows()}),
-               thrust::make_transform_iterator(gather_map.end<map_type>(),
-                                               index_converter<map_type>{source_table.num_rows()}),
-               ignore_out_of_bounds,
-               mr,
-               stream);
+    if (neg_indices == negative_indices::ALLOW) {
+      auto idx_converter = index_converter<map_type>{source_table.num_rows()};
+      return gather(source_table,
+                    thrust::make_transform_iterator(gather_map.begin<map_type>(), idx_converter),
+                    thrust::make_transform_iterator(gather_map.end<map_type>(), idx_converter),
+                    oob == out_of_bounds::IGNORE,
```

Comment:
Why not fix the bool parameter in the gather API being called here?

Reply:
I thought about that but that API is not in the `detail` namespace and will break clients code. If we are OK with that I can definitely change that API as well.